pandas/core/arrays/datetimes.py

```diff
@@ -145,8 +145,12 @@ def f(self):
                     kwds = freq.kwds
                     month_kw = kwds.get("startingMonth", kwds.get("month", 12))
 
+                if freq is not None:
+                    freqstr = to_offset(freq.freqstr).name
+                else:
+                    freqstr = freq
```

Comment:
I think this needs renaming too?

And in the `else` branch, you can just set it to `=None`

Reply:
thanks, I agree, here `freqstr` needs renaming too. I replaced it with `freq_name`